Implement email registration and password update

niklaslong opened the pull request:
Closes #6.

fancycade commented on lib/matrix_sdk/request.ex:
```diff
@@ -398,6 +595,37 @@ defmodule MatrixSDK.Request do
     }
 
+  @doc """
+  Gets a list of the third party identifiers the homeserver has associated with the user's account.
```
Looks you forgot to update this copy and paste in the doc comment.

niklaslong replied:
Good catch 👍

fancycade commented on lib/matrix_sdk/request.ex:
```diff
@@ -309,6 +309,116 @@ defmodule MatrixSDK.Request do
+  end
+
+  @doc """
+  Returns a `%Request{}` struct used to check the given address phone is not already associated with an account on the homeserver. This should be used to get a token to register a phone number as part of the initial user registration.
```
```suggestion
  Returns a `%Request{}` struct used to check the given phone number is not already associated with an account on the homeserver. This should be used to get a token to register a phone number as part of the initial user registration.
```

fancycade commented on lib/matrix_sdk/request.ex:
```diff
@@ -375,6 +485,93 @@ defmodule MatrixSDK.Request do
+      }
+  """
+  @spec deactivate_account(base_url, binary, map) :: t
+  def deactivate_account(base_url, token, opts \\ %{}),
```
```suggestion
  def deactivate_account(base_url, token, auth, opts \\ %{}),
```

I think auth is a required parameter here because of the type field.
https://matrix.org/docs/spec/client_server/latest#post-matrix-client-r0-account-deactivate

At least you do that with account_add_3pid. The spec seems to treat the auth param equally in both endpoints.

Granted the spec is a bit ambiguous.

niklaslong replied:
Actually I wonder if it's necessary in both endpoints at all—it might be optional, I'll investigate tomorrow.

niklaslong replied:
I'm inclined to treat `auth` as optional in both endpoints, given the example for the `account_add_3pid` endpoint. The auth API being interactive, the server response would make that explicit to the user in any case.

fancycade commented on lib/matrix_sdk/api.ex:
```diff
@@ -204,6 +285,14 @@ defmodule MatrixSDK.API do
     |> @http_client.do_request()
   end
 
```
Missing doc comment here

niklaslong commented:
@fancycade Thanks for the great review, it seems like I forgot to finish implementing `account_add_3pid`...

I've added the tests and treated `auth` as optional in both endpoints discussed above. I've corrected some doc issues and all in all, I'm hoping this should be good to go now.
